test/view/all.cpp

```diff
@@ -22,21 +22,27 @@ int main()
 
     int rgi[] = {1, 1, 1, 2, 3, 4, 4};
     std::vector<int> vi(begin(rgi), end(rgi));
-    std::list<int> li(begin(rgi), end(rgi));
+    std::list<int> const li(begin(rgi), end(rgi));
 
-    iterator_range<int *> x = view::all(rgi);
-    iterator_range<std::vector<int>::iterator> y = view::all(vi);
-    sized_iterator_range<std::list<int>::iterator> z = view::all(li);
+    auto x = view::all(rgi);
+    ::has_type<ref_view<decltype(rgi)> &>(x);
+    CONCEPT_ASSERT(range_cardinality<decltype(x)>::value == 7);
```

Comment:
Another advantage of `ref_view` over `iterator_range`: it preserves static extent.

Reply:
I suppose this is actually another side of `ref_view`'s single advantage over `subrange`: it preserves the range type instead of decaying to an iterator/sentinel pair.